emails/tests/views_tests.py

```diff
@@ -109,12 +112,76 @@
 )
 
 
+def create_email_from_notification(
+    notification: AWS_SNSMessageJSON, text: str
+) -> bytes:
+    """
+    Create an email from an SNS notification, return the serialized bytes.
+
+    The email will have the headers from the notification and the `text` value as the
+    plain text body.
+    """
+    message_json = notification.get("Message")
+    assert message_json
+    message = json.loads(message_json)
+    assert "mail" in message
+    mail_data = message["mail"]
+    assert "content" not in mail_data
+    email = EmailMessage()
+    assert "headers" in mail_data
+    for entry in mail_data["headers"]:
+        email[entry["name"]] = email[entry["value"]]
+    email.set_content(text)
+    return email.as_bytes()
+
+
+def assert_email_equals(output_email: str, name: str) -> None:
+    """
+    Assert the output equals the expected email, after replacements.
+
+    The MIME boundary strings have random content, are replaced with
+    "==[BOUNDARY#]==", with a number based on order found in the email.
```

Comment:
question (non-blocking): I don't understand why the MIME boundary strings have random content? Is that just how the `SESClient.send_raw_email` function works? And since we mock it, we "inherit" that behavior?

Reply:
[RFC 1521, section 7.2.1, The Content-Type Header Field](https://www.rfc-editor.org/rfc/rfc1521#section-7.2.1) is lengthy, but the key bit is:

> Encapsulation boundaries must not appear within the encapsulations,
>  and must be no longer than 70 characters, not counting the two
>  leading hyphens.

The Python library handles this by generating a large random value, then searching the email content for that value, and expanding it if found:

https://github.com/python/cpython/blob/22a8efa7408e00552c9845420caf925685dde92f/Lib/email/generator.py#L367-L388

This means the MIME boundaries change every time you run the test. Some Relay-generated boundaries looks like:

```
--===============3715842840874147262==
--===============5503079870118974321==
```

If you "view source" on your own emails, you'll see different methods used by different providers. Here's a few I found:

```
--91066fc9851403c96763f5d61c0a3917177cf1a64591a02b6e6bb5efb00b
--6ff9379f9f7fe9a87181d055163c0f4c5b4646e16fa0ad005c3144927afb
--=-fT/v9bIhfLoYa9WR+nQldA==
--9Ev1BlIFPznK=_?:
------=_Part_269670996_371849946.1695657797625
```

Reply:
Ah, that makes sense. With that in mind, I wrote this doc-block out to myself to try to make it more clear. Take it or leave it as another way to write this doc-block:

```
    Per RFC 1521, 7.2.1, MIME boundary strings must not appear in the bounded content.
    So python's email library generates a large random value for email boundary strings.
    So python makes different MIME boundary strings every time a test is run.
    So replace python's random boundary strings with a generic "==[BOUNDARY#]==",
    where the number is not random, but rather based on order found in the email.
```

Reply:
OK, expanding the doc string with this content.